cpp/src/io/json/reader_impl.cu

```diff
@@ -476,20 +476,17 @@ void reader::impl::set_data_types(device_span<uint64_t const> rec_starts,
         return std::find(std::cbegin(s), std::cend(s), ':') != std::cend(s);
       });
 
-    auto split_on_colon = [](std::string_view s) {
-      auto const i = s.find(":");
-      return std::pair{s.substr(0, i), s.substr(i + 1)};
```

Comment:
Just curious if changing this to `return std::make_pair(s.substr(0, i), s.substr(i + 1));` would fix this too?

Reply:
yup that also works! Its also less of a change - updated.